storage_models/src/common_structs/traits/storage_model.rs

```diff
@@ -0,0 +1,183 @@
+    ) -> Result<
+        AccountData<
+            EEVersion,
+            ObservableBytecodeHash,
+            ObservableBytecodeLen,
+            Nonce,
+            BytecodeHash,
+            BytecodeLen,
+            ArtifactsLen,
+            NominalTokenBalance,
+            Bytecode,
+        >,
+        SystemError,
+    >;
+
+    fn increment_nonce(
+        &mut self,
+        ee_type: ExecutionEnvironmentType,
+        resources: &mut Self::Resources,
+        address: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+        increment_by: u64,
+        oracle: &mut impl zk_ee::system_io_oracle::IOOracle,
+    ) -> Result<u64, UpdateQueryError>;
+
+    fn update_nominal_token_value(
+        &mut self,
+        from_ee: ExecutionEnvironmentType,
+        resources: &mut Self::Resources,
+        address: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+        update_fn: impl FnOnce(
+            &<Self::IOTypes as SystemIOTypesConfig>::NominalTokenValue,
+        ) -> Result<
+            <Self::IOTypes as SystemIOTypesConfig>::NominalTokenValue,
+            UpdateQueryError,
+        >,
+        oracle: &mut impl IOOracle,
+    ) -> Result<
+        <Self::IOTypes as zk_ee::types_config::SystemIOTypesConfig>::NominalTokenValue,
+        UpdateQueryError,
+    >;
+
+    fn get_selfbalance(
+        &mut self,
+        ee_type: ExecutionEnvironmentType,
+        resources: &mut Self::Resources,
+        address: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+    ) -> Result<
+        <Self::IOTypes as zk_ee::types_config::SystemIOTypesConfig>::NominalTokenValue,
+        SystemError,
+    >;
+
+    fn transfer_nominal_token_value(
+        &mut self,
+        from_ee: ExecutionEnvironmentType,
+        resources: &mut Self::Resources,
+        from: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+        to: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+        amount: &<Self::IOTypes as SystemIOTypesConfig>::NominalTokenValue,
+        oracle: &mut impl IOOracle,
+    ) -> Result<(), UpdateQueryError>;
+
+    fn deploy_code(
+        &mut self,
+        from_ee: ExecutionEnvironmentType,
+        resources: &mut Self::Resources,
+        at_address: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+        bytecode: &[u8],
+        bytecode_len: u32,
+        artifacts_len: u32,
+        oracle: &mut impl IOOracle,
+    ) -> Result<&'static [u8], SystemError>;
+
+    fn mark_for_deconstruction(
+        &mut self,
+        from_ee: ExecutionEnvironmentType,
+        resources: &mut Self::Resources,
+        at_address: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+        nominal_token_beneficiary: &<Self::IOTypes as SystemIOTypesConfig>::Address,
+        oracle: &mut impl IOOracle,
+    ) -> Result<(), SystemError>;
+
+    type Allocator: core::alloc::Allocator + Clone;
+    type InitData;
+
+    fn construct(init_data: Self::InitData, allocator: Self::Allocator) -> Self;
+
+    /// TODO
```

Comment:
Is this todo still relevant?

Reply:
Mostly as a reminder about "hacky" pubdata counting